software/due-date-changes.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import re
+import sys
+import time
+
+from datetime import datetime
+from jira import JIRA
+
+
+def dump(obj):
+    for attr in dir(obj):
+        print("obj.%s = %r" % (attr, getattr(obj, attr)))
+
+
+def print_help(stream=sys.stderr):
+    help_msg = """Usage: {0} [--projects] LIST,OF,PROJECTS
+"""
+    stream.write(help_msg.format(sys.argv[0]))
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-projects", 
+        help="Comma-separated list of project names (ie. HTCONDOR,SOFTWARE)",
+        default="HTCONDOR,SOFTWARE")
+    parser.add_argument("-o", 
+        help="Output file",
+        default=None)
+    args = parser.parse_args()
+
+    projects = args.projects
+    output_file = args.o
+
+    return {
+        "projects": projects,
+        "output_file": output_file
+    }
+
+
+def main():
+
+    # Parse arguments. Assume any error handling happens in parse_args()
+    try:
+        args = parse_args()
+    except Exception as err:
+        print(f"Failed to parse arguments: {err}", file=sys.stderr)
```

Comment:
So do you want an argparse failure to cause the script to exit?

Because it looks like it just continues here.

Also, it is good practice to handle the specific type of Exception you have in mind in a try/except.  `Exception` is a catchall for anything, including KeyboardInturrupt (Ctrl-C) or even SystemExit.  Not that there's a big window for other errors to sneak in here, but it is good practice to catch the type of errors you have in mind.

Reply:
It turns out this block is unnecessary, since any argpase failure will cause the script to exit directly in the parse_args() function. I'll remove it in a follow up.